src/update/UpdateManager.cpp

```diff
@@ -130,6 +130,42 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
 
                 if (code == wxID_YES)
                 {
+                    const audacity::network_manager::Request downloadRequest(mVersionPatch.download.ToStdString());
+                    auto downloadResponse = audacity::network_manager::NetworkManager::GetInstance().doGet(downloadRequest);
+
+                    downloadResponse->setRequestFinishedCallback([downloadResponse, this](audacity::network_manager::IResponse*) {
+                        mProgressDialog.reset();
+                        }
+                    );
+
+                    downloadResponse->setDownloadProgressCallback(
+                        [downloadResponse, this](int64_t current, int64_t expected) {
+
+                            static std::once_flag flag;
+                            std::call_once(flag, [this] {
+                                const auto title = XO("Audacity update")
+                                    .Translation();
+
+                                auto audacityPatchFilename = wxFileName(mVersionPatch.download).GetName();
```

Comment:
Wait, is mVersionPatch.download a url or is it a file name?  I'm confused now.

Reply:
Maybe you can get away with this, extracting the last path component.  It's a little hacky.

Reply:
It is a link to audacity installer file and have a two use cases here:
1 - Using like `std::string` URL for make request for CURL.
2 - Using like `wxString` for `wxFileName`, that contain the name of latest audacity patch.
May be I should create a new type for this cases like: `using PatchLink = wxString` for give a more anformation about `mVersionPatch.download` field.

Reply:
Well, leave it alone for now.

URLs use / as separator.  I was wondering whether this would work wrong on Windows where \ is the sperator, but you try to interpret a URL as a file path.  But if it tests all right, I won't insist on the type distinctions.

Reply:
yes, it work fine, I already tested it on Windows too.